util-core/src/test/java/com/twitter/util/FunctionCompilationTest.java

```diff
@@ -34,11 +36,66 @@ public Integer applyE() throws Exception {
         throw new Exception("Expected");
       }
     };
-    try {
-      fun.apply();
-      assert false : "Should have thrown";
-    } catch (Exception e) {
-      // pass: expected
-    }
+    fun.apply();
+  }
+
+  @Test
+  public void testMakeFunctionFromLambda() {
```

Comment:
is it OK that these are just testing compilation?  we might stop running them and just compile them at some point in time.